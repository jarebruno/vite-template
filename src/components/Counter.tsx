import { useState } from 'react'

export const Counter = () => {
  const [count, setCount] = useState(0)
  
  return (
    <button data-testid='counter-button' onClick={() => setCount((count) => count + 1)}>
      count is {count}
    </button>
  )
}