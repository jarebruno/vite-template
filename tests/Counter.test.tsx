import { render } from '@testing-library/react';
import { Counter } from '../src/components/Counter'

describe('Counter', () => {

  it('should render', () => {
    const { baseElement } = render(<Counter/>)
    expect(baseElement).toMatchSnapshot()
  });
})