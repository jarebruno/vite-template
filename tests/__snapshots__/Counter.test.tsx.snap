// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render 1`] = `
<body>
  <div>
    <button
      data-testid="counter-button"
    >
      count is 
      0
    </button>
  </div>
</body>
`;
